crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -316,6 +316,31 @@ pub(super) fn infer_unpack_types<'db>(db: &'db dyn Db, unpack: Unpack<'db>) -> U
     unpacker.finish()
 }
 
+/// Returns the type of the nearest enclosing class for the given scope.
+///
+/// This function walks up the ancestor scopes starting from the given scope,
+/// and finds the closest class definition.
+///
+/// Returns `None` if no enclosing class is found.a
+pub(crate) fn enclosing_class_symbol<'db>(
```

Comment:
Orthogonal to this PR, but this should really be renamed to `enclosing_class_type` -- it returns a `Type`, not a `Symbol`.